discmoji/bot.py

```diff
@@ -29,22 +29,53 @@
 from .types import OPCODES
 from .intents import BotIntents
 from .context import Invoked
+
+class CommandManager:
+    def __init__(self, bot: 'Bot'):
+        self.bot = bot
+        self._all_cmds: List[Command] = []
+
+    def command(self, name: str):
+        """A decorator that registers a command with the specified name."""
+        def decor(func: Callable):
+            self._all_cmds.append(Command(name=name))
+            return Command(name=name)
+        return decor
+
+class GuildManager:
+    def __init__(self, bot: 'Bot'):
```

Comment:
this is another way of annotating. you can use strings to annotate an argument and this will show up in an ` __annotations__ ` attribute ( if it existed somewhere in 3.13 and 3.14 ). this is also, and the only, way to annotate arguments if the annotation is derived from the ` typing.TYPE_CHECKING ` import lock

Reply:
ohhh ty